core/input/input.cpp

```diff
@@ -706,16 +706,16 @@ void Input::_parse_input_event_impl(const Ref<InputEvent> &p_event, bool p_is_em
 						if (!action.axis_pressed) {
 							action.pressed++;
 							action.axis_pressed = true;
+							if (action.pressed == 1) {
+								action.pressed_physics_frame = Engine::get_singleton()->get_physics_frames();
+								action.pressed_process_frame = Engine::get_singleton()->get_process_frames();
+							}
 						}
 					} else {
 						action.pressed++;
```

Comment:
With these changes, in this else-branch the pressed-frame is now no longer set. This would lead to missing pressed-frame info's for Actions that are caused by keyboard events.

Reply:
Woops, my bad.
I fixed it with new flag is_new_press and returned the pressed frame assignments the original scope.

Tested for both axis, and keyboard input. Both presses and "just_pressed" functionality work well now.